libnavui-androidauto/src/androidTest/java/com/mapbox/androidauto/car/navigation/lanes/CarLanesImageRendererTest.kt

```diff
@@ -39,6 +41,25 @@ class CarLanesImageRendererTest {
         background = Color.RED
     )
 
+    @Test
+    fun cache_test() {
+        val lane = mockk<Lane> {
+            every { allLanes } returns listOf(
+                mockk {
+                    every { drivingSide } returns "right"
+                    every { activeDirection } returns "uturn"
+                    every { isActive } returns true
+                    every { directions } returns listOf("uturn")
+                }
+            )
+        }
+        val img1 = carLanesImageGenerator.renderLanesImage(lane)
+        val img2 = carLanesImageGenerator.renderLanesImage(lane)
+
+        assertNotNull(img1)
+        assertSame(img1, img2)
+    }
```

Comment:
This is not testing the actual equality since it's the same object.

It may be better to add build the data objects, also test the cache miss case

```
    @OptIn(ExperimentalMapboxNavigationAPI::class)
    @Test
    fun cache_hit_test() {
        val lane1 = LaneFactory.buildLane(
            allLanes = listOf(LaneIndicator.Builder()
                .drivingSide("right")
                .activeDirection("uturn")
                .isActive(true)
                .directions(listOf("uturn"))
                .build())
        )
        val lane2 = LaneFactory.buildLane(
            allLanes = listOf(LaneIndicator.Builder()
                .drivingSide("right")
                .activeDirection("uturn")
                .isActive(true)
                .directions(listOf("uturn"))
                .build())
        )

        val img1 = carLanesImageGenerator.renderLanesImage(lane1)
        val img2 = carLanesImageGenerator.renderLanesImage(lane2)

        assertNotNull(img1)
        assertSame(img1, img2)
    }

    @OptIn(ExperimentalMapboxNavigationAPI::class)
    @Test
    fun cache_miss_test() {
        val lane1 = LaneFactory.buildLane(
            allLanes = listOf(LaneIndicator.Builder()
                .drivingSide("right")
                .activeDirection("uturn")
                .isActive(true)
                .directions(listOf("uturn"))
                .build())
        )
        val lane2 = LaneFactory.buildLane(
            allLanes = listOf(LaneIndicator.Builder()
                .drivingSide("right")
                .activeDirection("uturn")
                .isActive(true)
                .directions(listOf("straight"))
                .build())
        )

        val img1 = carLanesImageGenerator.renderLanesImage(lane1)
        carLanesImageGenerator.renderLanesImage(lane2)
        val img3 = carLanesImageGenerator.renderLanesImage(lane1)

        assertNotNull(img1)
        assertSame(img1, img3)
    }
```

Reply:
But the cache miss case is covered by the other tests, but this will fail even though the values are the same

```
    @Test
    fun cache_test() {
        val lane1 = mockk<Lane> {
            every { allLanes } returns listOf(
                mockk {
                    every { drivingSide } returns "right"
                    every { activeDirection } returns "uturn"
                    every { isActive } returns true
                    every { directions } returns listOf("uturn")
                }
            )
        }
        val lane2 = mockk<Lane> {
            every { allLanes } returns listOf(
                mockk {
                    every { drivingSide } returns "right"
                    every { activeDirection } returns "uturn"
                    every { isActive } returns true
                    every { directions } returns listOf("uturn")
                }
            )
        }
        val img1 = carLanesImageGenerator.renderLanesImage(lane1)
        val img2 = carLanesImageGenerator.renderLanesImage(lane2)

        assertNotNull(img1)
        assertSame(img1, img2)
    }
```

Reply:
Great idea. I've updated `CarLanesImageRendererTest` to include both tests.